connect/clients/nats.py

```diff
@@ -299,6 +301,10 @@ async def stop_nats_clients():
     global nats_retransmit_canceled
     nats_retransmit_canceled = True
 
+    retransmit_task.cancel()
+    tasks = [retransmit_task]
+    await asyncio.gather(*tasks)
```

Comment:
are we gathering here to ensure that our cancel call is "blocking"?

Reply:
It's just a helper to wait for the cancelled task to finish.  We could have used the lower-level asyncio.wait here as well.